### PR TITLE
Support partial dictionaries

### PR DESCRIPTION
**Pull request by lukebjerring**
Fixes https://github.com/w3c/web-platform-tests/issues/5714

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/web-platform-tests/10159)

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PASSED

Started: 2018-03-23 17:50:08
Finished: 2018-03-23 17:57:19





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/26405)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/357510275)

File: resources/test/tests/idlharness/IdlDictionary/test_partial_dictionary.html [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+}, 'Partial dictionaries');
+
+test(() => {
+  try {
+    let idlArray = new IdlArray();
+    idlArray.add_idls('partial dictionary D {};');
+    idlArray.test();
+  } catch (e) {
+    return;
+  }
```
Comment (lukebjerring) [inline resources/test/tests/idlharness/IdlDictionary/test_partial_dictionary.html]: try/catch code-smell covered by https://github.com/w3c/web-platform-tests/issues/10162

File: resources/idlharness.js [hunk 2]
```diff
@@ -257,7 +257,8 @@ IdlArray.prototype.internal_add_idls = function(parsed_idls, options)
 
     parsed_idls.forEach(function(parsed_idl)
     {
-        if (parsed_idl.type == "interface" && parsed_idl.partial)
+        if (parsed_idl.partial
+            && ["interface", "dictionary"].find(t => parsed_idl.type == t))
```
Comment (foolip) [inline resources/idlharness.js]: Optional simplification: `["interface", "dictionary"].includes(parsed_idl.type)` would do the trick here too.

**Review by foolip — APPROVED** [review]
Thanks! Added test coverage is great :)

File: resources/idlharness.js (hunk 2 above)
Comment (lukebjerring) [reply resources/idlharness.js]: lol. tried 'contains' and fell back to something I knew. Changed.

**Pull request merged**